### PR TITLE
InterruptedException should not be ignored

### PR DESCRIPTION
**Pull request by asia-zengtao**
InterruptedException should not be ignored  
print InterruptedException  message and  Restore interrupted state

**Make sure set the target branch to `develop`**

## What is the purpose of the change

InterruptedException should not be ignored

## Brief changelog

print InterruptedException  message and  Restore interrupted state

## Verifying this change

try {
        scheduledExecutorService.awaitTermination(5000, TimeUnit.MILLISECONDS);
 } catch (InterruptedException ignore) {
        BrokerController.LOG.warn("shutdown ScheduledExecutorService was Interrupted!  ", ignore);
        Thread.currentThread().interrupt();
 }

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by mxsm** [comment]
Hi @asia-zengtao  you should create issue first, This issue describes the problem and explains why. then submit a PR to fixed, you can see this docmuent how the PR related issue.
https://docs.github.com/cn/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by hzh0425** [comment]
Welcome !

**Comment by lizhanhui** [comment]
InterruptedException is a key approach to enhance program responsiveness... As [this article](https://www.baeldung.com/java-interrupted-exception) points out, different ways of handling are required according to specific use cases. Unfortunately, most interrupted exceptions in the codebase are simply ignored, leaving a major place to improve. 

Glad to see that you point this issue out and attempt to improve.

**Review by tsunghanjacktsai — APPROVED** [review]
Hi @asia-zengtao ,

Great work and welcome to the RocketMQ community. Hope you could keep making our codebase better.

**Review by hzh0425 — DISMISSED** [review]

File: broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java [hunk 1]
```diff
@@ -1423,6 +1423,8 @@ protected void shutdownScheduledExecutorService(ScheduledExecutorService schedul
         try {
             scheduledExecutorService.awaitTermination(5000, TimeUnit.MILLISECONDS);
         } catch (InterruptedException ignore) {
+            BrokerController.LOG.warn("shutdown ScheduledExecutorService was Interrupted!  ", ignore);
+        	Thread.currentThread().interrupt();
```
Comment (mxsm) [inline broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java:1427]: https://github.com/apache/rocketmq/actions/runs/3449152754/jobs/5767526841
Error:  /home/runner/work/rocketmq/rocketmq/broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java:1427:9: File contains tab characters (this is the first instance). [FileTabCharacter]
Error:  /home/runner/work/rocketmq/rocketmq/broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java:1427:17: 'catch' child has incorrect indentation level 16, expected level should be 12. [Indentation]

Here has some problems need to fix in CI.

**Review by mxsm — CHANGES_REQUESTED** [review]
Code format needs modify.

**Review by tsunghanjacktsai — APPROVED** [review]

**Review by lizhanhui — APPROVED** [review]

**Pull request merged**